autoload/coverage.vim

```diff
@@ -93,31 +93,41 @@ endfunction
 
 ""
 " Toggles coverage layer.
-function! s:CoverageToggle() abort
+function! s:CoverageToggle(skip_cache) abort
   if has_key(s:visible, expand('%:p'))
     call s:CoverageHide()
   else
-    call s:CoverageShow()
+    call s:CoverageShow(a:skip_cache)
   endif
 endfunction
 
 
 ""
 " Shows coverage layer. If [explicit_provider] is set, it will be used for
 " fetching the coverage data.
-function! s:CoverageShow(...) abort
+function! s:CoverageShow(skip_cache, ...) abort
   let l:filename = expand('%:p')
-  if !has_key(s:visible, l:filename)
-    if (has_key(s:cache, l:filename))
-      call s:RenderFromCache(l:filename)
+
+  if has_key(s:visible, l:filename)
+    if a:skip_cache
+      call s:CoverageHide()
     else
-      if a:0 > 0
-        call coverage#ShowCoverage(maktaba#ensure#IsString(a:1))
-      else
-        call coverage#ShowCoverage()
-      endif
+      " This file is already being shown, no need to re-show.
+      return
     endif
   endif
+
+  if has_key(s:cache, l:filename) && !a:skip_cache
+    call s:RenderFromCache(l:filename)
+    return
+  endif
+
+  if a:0 > 0
+    call coverage#ShowCoverage(maktaba#ensure#IsString(a:1))
```

Comment:
Could you give this `a:1` a name, now that there are more args to keep track of?
```vim
  let l:provider = maktaba#ensure#IsString(a:1)
  call coverage#ShowCoverage(l:provider)
```